naima/models.py

```diff
@@ -431,3 +431,76 @@ def __call__(self, e):
         e = _validate_ene(e)
         interpy = np.power(10, self._interplogy(np.log10(e.to('eV').value)))
         return self.amplitude * interpy * self.unit
+
+
+class EblAbsorptionModel(TableModel):
+    """
+    A TableModel containing the different absorption values from a specific model.
+
+    It returns dimensionless opacity values, that could be multiplied to any model.
+
+    Parameters
+    ----------
+    redshift : float
+        Redshift considered for the absorption evaluation.
+    ebl_absorption_model : string
+        Name of the EBL absorption model to use (Dominguez by default).
+
+    See Also
+    --------
+    TableModel
+
+    Notes
+    -----
+    Dominguez model refers to the Dominguez 2011 EBL model. Current implementation 
+    does NOT perform an interpolation in the redshift, so it just uses the closest
+    z value from the finely binned tau_dominguez11.out file (delta_z=0.01).
+    """
+
+    def __init__(self, redshift, ebl_absorption_model='Dominguez'):
+        from scipy.interpolate import interp1d
+        from astropy.utils.data import get_pkg_data_filename
+        import os
+
+        # check that the redshift is a positive scalar
+        if isinstance(redshift, u.Quantity):
+            self.redshift = validate_scalar('redshift',
+                                            redshift,
+                                            domain='positive',
+                                            physical_type='dimensionless')
+        elif not np.isscalar(redshift) or redshift < 0:
+            raise ValueError("Redshift should be a positive scalar value.")
+        else:
+            self.redshift = redshift * u.dimensionless_unscaled
```

Comment:
You should run the quantity through `validate_scalar` after adding the `dimensonless_unscaled` units.
Maybe something like (pseudocode):
```
if not isinstance(redshift, u.Quantity):
    redshift *= u.dimensionless_unscaled

self.redshift = validate_scalar(...)
```
(validate scalar will check that it is scalar and positive, so no need to do it separately)